ygen/genstate.go

```diff
@@ -52,6 +52,9 @@ type MappedType struct {
 	// the generated code from the structs maintains only type validation,
 	// this is not currently a limitation.
 	UnionTypes map[string]int
```

Comment:
Should we refactor here to have `UnionTypes` be `map[string]MappedUnionSubType` and then make the `int` that is the value here be within that value? We could do it the other way around by having the `UnionTypeInfo` just store the int now and then we can remove this in a subsequent PR?

Reply:
Sounds good, do you mind if I do
1) Add the index to `UnionTypeInfo`
2) Remove `UnionTypes`
both in a later PR after the chain?

Reply:
@robshakir  Done, I put it at the end of the chain here: https://github.com/openconfig/ygot/pull/684